### ISSUE TITLE
[Request] make SDR-Server work with HackRF and/or SoapySDR

### ISSUE THREAD

**Issue by jwcrawley** [dernasherbrezon/sdr-server#6]
Greetings, 

I would LOVE if I could take this SDR-server over to Pluto's, HackRF's, BladeRF's, and more. And especially so, since these SDR's can generate a much wider waveform and would greatly benefit to splitting them up and processing remotely.

**Comment by nmaster2042** [comment]
Hi.

Completely agree with this.

Extend sdr-server to other SDRs would be a great add.

**Comment by wgaylord** [comment]
I would be interested in this as well, looking into making a fork for it. But I have no experence at all with writing software using soapy or SDR stuff in C.

**Comment by dernasherbrezon** [comment]
I can work on that, but HackRF has much wider bandwidth - 20mhz. So to make sdr-server useful it needs to be even more faster. Or run on a powerful hardware.

**Comment by Strontvlieg** [comment]
Yeah hackrf +1 
I think an old HP G2 800 I5-6500 mini is fast enough to handle it. And with a good antenna and the hackrf Mount on the antenna outside it cloud not better

**Comment by wgaylord** [comment]
I will say adding soapysdr support would open up sdr-server to using basically any sdr that soapy supports now and in the future, so that would be a great enhancement (since I was interested in using it with my LimeSDR Mini)

**Comment by nmaster2042** [comment]
Yes I agree SoapySDR would bring support of Hackrh and many other SDRs.

**Comment by dernasherbrezon** [comment]
Added support for HackRF. Tested the following scenario:

- band_sampling_rate=19200000 (19.2Mhz)
- 2 clients. each asking for 48khz (./sdr_server_client -f 100000000 -p 8090 -s 48000 ouptut.cf32)
- SIMD optimisation detected: 

```
SIMD optimization: AVX
compilation flags: -march=native -mavx -O3 -DNDEBUG -ffast-math
```

CPU usage is 75% on "Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz".

I will look further into fixed point math optimisations. But for now consider using smaller band_sampling_rate.
